jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.kt

```diff
@@ -56,6 +56,7 @@ private fun notifyNewCustomization(project: Project) {
     )
 }
 
+// TODO: better to split this component into two parts, of which one is application level component and one is project level component in the long run
```

Comment:
just like how ToolkitAuthManager and ToolkitConnectionManager work together

Reply:
I still think CodeWhisperer should be an application service overall not project one.

Reply:
yes, so I think like how ToolkiAuth & ToolkitConnection managers work, the drawback of current implementation is that the customization selection is bound to "connection" level, it's not easy to make it "project" level binding with current implementation. But I will take this as follow-up and it's not urgently required for now.

Reply:
My idea is that 
App level component will manage the key-value pair of conneciton to customization options snapshot, validation etc. 


Project level component will simply manage the selection of customization.

Reply:
sure,
but my point is I would just make CW connection a app-level connection.

Reply:
I see

Reply:
> but my point is I would just make CW connection a app-level connection.
that would make things easier